lib/evmone/execution_state.hpp

```diff
@@ -172,6 +172,7 @@ class ExecutionState
 
 private:
     evmc_tx_context m_tx = {};
+    std::optional<std::unordered_map<evmc::bytes32, bytes_view>> m_initcodes;
```

Comment:
Will it fly if we omit `std::optional` (empty map meaning no initcodes)?

Reply:
I think this is optimization to avoid calling `get_tx_context().initcodes` on every `get_tx_initcode_by_hash()`.

So it will work without optional, but for no initcodes will check `tx_context.initcodes_count` on every TXCREATE (will not query the host though, thanks to cached `tx_context`).